src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.hidden === false) {
+      animal.health -= 50;
+    }
+
+    if (animal.health <= 0) {
+      Animal.alive = Animal.alive.filter((a) => a !== animal);
```

Comment:
[CODE PERFORMANCE] The current implementation of the `bite` method uses a nested check (if statement within another if statement) to determine if an animal should be removed from the `Animal.alive` array. It is more performant to use a single filter operation after the health deduction to remove any dead animals. This avoids unnecessary checks and potential performance issues with larger arrays of animals.